### PR TITLE
USHIFT-1337: Add support for creating N virtual machines

### PR DESCRIPTION
**Pull request by pacevedom**
Add support for arbitrary number of VMs to be created. Data about ssh port, user and kubeconfig for each VM is stored in the shared dir for later use.

### CODE REVIEW

File: ci-operator/step-registry/openshift/microshift/infra/iso-boot/openshift-microshift-infra-iso-boot-commands.sh [hunk 1]
```diff
@@ -48,11 +56,12 @@ sudo /sbin/iptables -t nat -I PREROUTING -p tcp --dport "${API_EXTERNAL_PORT}" -
 sudo /sbin/iptables -I FORWARD -o virbr0 -p tcp -d \${VMIPADDR} --dport 22 -j ACCEPT
 sudo /sbin/iptables -t nat -I PREROUTING -p tcp --dport "${SSH_EXTERNAL_PORT}" -j DNAT --to \${VMIPADDR}:22
 EOF
-chmod +x /tmp/boot.sh
+  chmod +x /tmp/boot.sh
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/infra/iso-boot/openshift-microshift-infra-iso-boot-commands.sh:59]: Unnecesary space?

File: ci-operator/step-registry/openshift/microshift/infra/iso-boot/openshift-microshift-infra-iso-boot-commands.sh [hunk 2]
```diff
@@ -23,12 +23,20 @@ Host ${IP_ADDRESS}
...
 
-cat <<EOF > /tmp/boot.sh
+# number of VMs to create. This will change over time as we add more tests.
+NUM_VMS=1
+echo "${NUM_VMS}" > ${SHARED_DIR}/num_vms
+for (( i=0; i<$NUM_VMS; i++ ))
+do
+  API_EXTERNAL_PORT=$((API_EXTERNAL_BASE_PORT+$i))
+  SSH_EXTERNAL_PORT=$((SSH_EXTERNAL_BASE_PORT+$i))
+  VM_NAME="ushift_${i}"
```
[the remark this answers is not in the document]
Comment (pacevedom) [reply ci-operator/step-registry/openshift/microshift/infra/iso-boot/openshift-microshift-infra-iso-boot-commands.sh]: Done

**Pull request merged**